<launch>

  <node pkg="image_proc" type="image_proc"
        ns="front_cam"
        name="image_proc">
  </node>

  <arg name="camera" default="front_cam"/>
  <arg name="message" default="/image_rect_color"/>
  <arg name="center_offset" default="false"/>
  <arg name="x_offset" default="0"/>
  <arg name="y_offset" default="128"/>
  <arg name="width" default="1920"/>
  <arg name="height" default="540"/>
  <arg name="resize_gain" default="1"/>
  <arg name="rate" default="1.0"/>
  <arg name="enable" default="true"/>
  <arg name="rotate" default="false"/>
  <arg name="mono" default="false"/>
  <arg name="jpeg_quality" default="80"/>

  <node pkg="pharos_image_transport" type="pretreat" name="$(arg camera)_pretreat" output="screen">
    <param name="camera" value="$(arg camera)"/>
    <param name="message" value="$(arg message)"/>

    <param name="rate" value="$(arg rate)"/>

    <param name="center_offset" value="$(arg center_offset)"/>
    <param name="x_offset" value="$(arg x_offset)"/>
    <param name="y_offset" value="$(arg y_offset)"/>
    <param name="crop_width" value="$(arg width)"/>
    <param name="crop_height" value="$(arg height)"/>
    <param name="resize_gain" value="$(arg resize_gain)"/>

    <param name="enable" value="$(arg enable)"/>
    <param name="rotate" value="$(arg rotate)"/>
    <param name="mono" value="$(arg mono)"/>
    
    <param name="jpeg_quality" value="$(arg jpeg_quality)"/>

    <remap from="example" to="pretreat_param"/>
  </node>

</launch>
